### PR TITLE
Fix/staking

### PR DESCRIPTION
**Pull request by Divya-Solulab**
fix logic to consider the service active and eligible for rewards, even if it joins in between a liveness period

### CODE REVIEW

**Review by 0xArdi — APPROVED** [review]

File: packages/valory/skills/liquidity_trader_abci/rounds.py [hunk 1]
```diff
@@ -216,8 +216,17 @@ def end_block(self) -> Optional[Tuple[BaseSynchronizedData, Enum]]:
         if synced_data.service_staking_state == StakingState.EVICTED.value:
             return synced_data, Event.SERVICE_EVICTED
 
-        if synced_data.most_voted_tx_hash is None:
-            return synced_data, Event.NEXT_CHECKPOINT_NOT_REACHED_YET
+        if (
+            synced_data.most_voted_tx_hash is None
+            and synced_data.is_staking_kpi_met == False
```
Comment (Adamantios) [inline packages/valory/skills/liquidity_trader_abci/rounds.py:221]:
```suggestion
            and synced_data.is_staking_kpi_met is False
```

File: packages/valory/skills/liquidity_trader_abci/payloads.py [hunk 2]
```diff
@@ -42,7 +43,7 @@ class CheckStakingKPIMetPayload(BaseTxPayload):
     """A transaction payload for the CheckStakingKPIMetRound."""
 
     tx_submitter: str
-    kpi_met_for_the_day: Optional[bool]
+    is_staking_kpi_met: Optional[bool]
```
Comment (Adamantios) [inline packages/valory/skills/liquidity_trader_abci/payloads.py:46]: Now that there are so many common attributes you can convert `CallCheckpointPayload` to:

```python
@dataclass(frozen=True)
class CallCheckpointPayload(BaseTxPayload):
    """A transaction payload for the CallCheckpointRound."""

    service_staking_state: int
    min_num_of_safe_tx_required: Optional[int]
```

File: packages/valory/skills/liquidity_trader_abci/rounds.py [hunk 3]
```diff
@@ -216,8 +216,17 @@ def end_block(self) -> Optional[Tuple[BaseSynchronizedData, Enum]]:
...
+        ):
+            return synced_data, Event.STAKING_KPI_NOT_MET
+
+        if (
+            synced_data.most_voted_tx_hash is None
+            and synced_data.is_staking_kpi_met == True
+        ):
+            return synced_data, Event.STAKING_KPI_MET
```
Comment (Adamantios) [inline packages/valory/skills/liquidity_trader_abci/rounds.py:229]:
```suggestion
        if (
            synced_data.most_voted_tx_hash is None
            and synced_data.is_staking_kpi_met is True
        ):
            return synced_data, Event.STAKING_KPI_MET
```

File: packages/valory/skills/liquidity_trader_abci/rounds.py
Comment (Adamantios) [inline packages/valory/skills/liquidity_trader_abci/rounds.py:1]: In `CallCheckpointRound`, what if both `is_staking_kpi_met` and `most_voted_tx_hash` are `None`? Shouldn't we re-enter `CallCheckpointRound` instead?

**Review by Adamantios — APPROVED** [review]

**Pull request merged**